pkg/gofr/datasource/file/fs.go

```diff
@@ -61,3 +61,36 @@ func (fileSystem) RemoveAll(path string) error {
 func (fileSystem) Rename(oldname, newname string) error {
 	return os.Rename(oldname, newname)
 }
+
+// Stat returns the file/directory info.
+func (fileSystem) Stat(name string) (FileInfo, error) {
+	return os.Stat(name)
+}
+
+// Getwd returns the full path of the current working directory.
+func (fileSystem) Getwd() (string, error) {
+	return os.Getwd()
+}
+
+// ChDir changes the current working directory to the named directory.
+// If there is an error, it will be of type *PathError.
+func (fileSystem) ChDir(dir string) error {
+	return os.Chdir(dir)
+}
+
+// ReadDir reads the named directory, returning all its directory entries sorted by filename.
+// If an error occurs reading the directory, ReadDir returns the entries it was able to read before the error, along with the error.
+// It returns the list of files/directories present in the current directory when "." is passed.
+func (fileSystem) ReadDir(dir string) ([]FileInfo, error) {
+	entries, err := os.ReadDir(dir)
+
+	fileInfo := make([]FileInfo, len(entries))
+	for i := range entries {
+		fileInfo[i], err = entries[i].Info()
+		if err != nil {
+			return fileInfo, err
```

Comment:
I would return nil, but it's open for debate

```suggestion
			return nil, err
```

Reply:
If an error occurs reading the directory, ReadDir returns the entries it was able to read before the error, along with the error. Hence I have returned fileInfo instead of nil. I have added a comment about it.